dallinger/recruiters.py

```diff
@@ -77,6 +81,11 @@ class PsiTurkRecruiter(Recruiter):
 
     def __init__(self):
         """Set up the connection to MTurk and psiTurk web services."""
+
+        # commit to the db so if anything goes wrong the rest of the
+        # request is safe
+        session.commit()
```

Comment:
@suchow ok, I suggest that we have custom commit() whenever it is about to create the recruiter, this will have the same effect, but the committing will be in custom.py and so more explicit.

This won't stop negative consequences of unforeseen errors outside of the recruiter. The only way to do that would be to commit after every change to the db which is a bit ugly, and probably time consuming.

As a note, while I can see the benefit of having explicit commits, this goes against previous Wallace/Dallinger style which was to hide all this stuff from the user: We managed to set it up so that objects don't need to be manually added to the db. Moreover, all routes commit to the db at their completion and because everything occurs as part of a request this means that (assuming nothing goes wrong within a route) the experimenter never needs to commit changes to the db either.